src/WrapExternStages.cpp

```diff
@@ -89,8 +89,9 @@ class WrapExternStages : public IRMutator2 {
                 // Make the call to upgrade old buffer
                 // fields into the original new
                 // buffer. Important for bounds queries.
+                const int bounds_query_only = a.kind == Argument::InputBuffer ? 1 : 0;
```

Comment:
This appears to assume that bounds queries only happen for inputs. Bounds queries to define_extern stages are done in two phases. In the first the output buffer is being queried (e.g. in case the extern stage requires an output width of at least 8). In the second the input buffers are being queried.

Reply:
Yes; the gotcha is that for the legacy_buffers case, the output buffers are always copied in and out thru stack temporaries, so there is no data race, while input buffers aren't.

(I may just fix the places that I know of that use parallel define-extern to not use legacy buffers, then declare this code old and broken anyway...)

Reply:
I'm OK with landing this.